spec/compiler/semantic/exception_spec.cr

```diff
@@ -145,11 +145,19 @@ describe "Semantic: exception" do
   end
 
   it "errors if caught exception is not a subclass of Exception" do
-    assert_error "begin; rescue ex : Int32; end", "Int32 is not a subclass of Exception"
+    assert_error "begin; rescue ex : Int32; end", "Int32 is not a module type or subclass of Exception"
+  end
+
+  it "errors if caught exception is a union but not all types are valid" do
+    assert_error "begin; rescue ex : Union(Exception, String); end", "Not all union members of (Exception | String) are a module type or subclass of Exception"
+  end
+
+  it "errors if caught exception is a nested union but not all types are valid" do
+    assert_error "begin; rescue ex : Union(Exception, Union(Exception, String)); end", "Not all union members of (Exception | String) are a module type or subclass of Exception"
```

Comment:
Compiler seems to actually de-dupe+flatten this union, hence why it passes; but the question is should the error message include the full union as it does now, or only the nested union that is invalid?